File: 148.排序链表.go
/*
 * @lc app=leetcode.cn id=148 lang=golang
 *
 * [148] 排序链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func sortList(head *ListNode) *ListNode {
	temp1 := head
	temp2 := temp1
	res := []int{}
	for head != nil {
		res = append(res, head.Val)
		head = head.Next
	}
	sort.Ints(res)
	for _, one := range res {
		temp2.Val = one
		temp2 = temp2.Next
	}
	return temp1
}

// @lc code=end

File: 148.排序链表.py
#
# @lc app=leetcode.cn id=148 lang=python3
#
# [148] 排序链表
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp1 = temp2 = head
        res = []        # 用一个列表 res 写入所有的元素，排序后依次覆盖原值
        while head:
            res.append(head.val)
            head = head.next
        res  = sorted(res)  # 排序
        for one in res:
            temp2.val = one
            temp2 = temp2.next
        return temp1
# @lc code=end

